Js_capston_project

vic778 opened the pull request:
- Use TVMaze API to get a list of movies with a unique movie id and detailed information about the movie
- Use Involvement API to record the different user interactions (likes and comments)
- The movie has the like button that allows users to like the movie and the no. of likes is updated on the screen
- The movie has a comment section that allows users to add a new comment, view comments, and count the no. of comments written under a given movie
- Test the movie and comment counter with Jest
- Update the header and the footer
- This is the [Video](https://drive.google.com/drive/folders/1IYtVwRFl-fu-Ww02gMjvGZRUW_NLOW62?usp=sharing) in regards to our project

cwaku committed:
Setup project

cwaku committed:
Update REAME.md and fix linter error

cwaku committed:
fix linter error

vic778 committed:
Added sass

vic778 committed:
Added dist

vic778 committed:
Fixed linter error

vic778 committed:
Added bootsrrap

vic778 committed:
Created list-item div to displaying items

vic778 committed:
Imported show function to display in render

vic778 committed:
Added screens

vic778 committed:
Creted function to fetch item from the Api

vic778 committed:
Exported show function and initialized araayOfShow variable to get items fron the Url

vic778 committed:
fixed linter error

vic778 committed:
Removed Import bootstrap

vic778 committed:
Initialized the project

vic778 committed:
Describe the project

vic778 committed:
Added footer section

cwaku committed:
Merge pull request #29 from vic778/Display

Display data from API

vic778 committed:
Crretaed a function to show the number of movies we have in the db and and how much is already seen by the user

vic778 committed:
Update README.md

cwaku committed:
Merge pull request #31 from vic778/Counter

Counter

cwaku committed:
Show comment popup

cwaku committed:
Display selected item's details on popup

vic778 committed:
Merge pull request #32 from vic778/commentPopup

Comment popup

cwaku committed:
Add API calls to POST and GET comments and counts

cwaku committed:
Add logic to add and display comments and counts

cwaku committed:
Merge pull request #33 from vic778/displayComments

Display comments

vic778 committed:
Create a function to test the movisCounter

vic778 committed:
Update README.md

vic778 committed:
Update README.md

cwaku committed:
Add tests for comments counter

cwaku committed:
Merge pull request #34 from vic778/MovieCounterJestTest

Create a function to test the movisCounter

vic778 committed:
Merge pull request #35 from vic778/testComments

Add tests for comments counter

cwaku committed:
Update README.md

cwaku committed:
Merge pull request #36 from vic778/updateReadme

Update README.md

vic778 committed:
Update README.md

vic778 committed:
Update README.md

Added the live Demo Url

cwaku committed:
Add deployment config

TedLivist commented on src/scripts/Display/cards.js:
```diff
@@ -0,0 +1,53 @@
+      clone.classList.remove('d-none');
+      const footer = clone.querySelector('.card-footer');
+      const commentBtn = footer.querySelector('a');
+      commentBtn.addEventListener('click', () => {
+        const inputForm = document.querySelector('.form');
+        const viewCommentsBtn = document.querySelector('.display-comment-btn');
+        inputForm.dataset.id = show.id;
+        viewCommentsBtn.dataset.id = show.id;
```
**Remember to reflect your changes in the live demo too. For both points, I have added a Loom Video link for better elaboration (below):**

- [ ] I think your comments modal implementation has a bug. When I comment on a movie and retrieve the comments, and go to another movie modal I see comments from the previous movie, but that should not be the case, as stated in requirements (see image). Kindly fix that. It is merely my suggestion that you call your `displayCommentCount()` and `displayAllComments()` functions upon modal load, depending on how your code is implemented.

  ![ricky-comm](https://user-images.githubusercontent.com/36542986/148204748-8d8fb9cc-f1b0-4ba5-9862-0b5b5595d42c.jpg)

- [x] Also, after navigating from a movie modal that has comments to one that does not I still see the comments count from the previous movie modal even after clicking **`View comments`** button. Kindly fix that.

**Loom Video Link: [here](https://loom.com/share/1027ae26b6474d129777b9fdb0729e4c)**

TedLivist reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @vic778 and @cwaku,

Good job so far!
✔️ Good documentation
✔️ Nice modularization of code
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

cwaku committed:
Fix comment and counter display

vic778 committed:
Merge pull request #38 from vic778/fixComments

Fix comment and counter display

arslanbisharat reviewed (approved):
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 1,707.90kb -> 1,468.72kb (14%)

/img/screen3.png -- 280.53kb -> 208.68kb (25.61%)
/img/screen2.png -- 735.80kb -> 645.85kb (12.23%)
/img/screen1.png -- 691.57kb -> 614.18kb (11.19%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

vic778 committed:
Merge pull request #40 from vic778/imgbot

[ImgBot] Optimize images
